src/v8/v8.cc

```diff
@@ -263,9 +268,45 @@ bool V8::load(const std::string &code, bool allow_precompiled) {
     assert((shared_module_ != nullptr));
   }
 
+  buildFunctionNameIndex();
+
   return module_ != nullptr;
 }
 
+void V8::buildFunctionNameIndex() {
+  // build function index -> function name map for backtrace
+  // https://webassembly.github.io/spec/core/appendix/custom.html#binary-namesubsection
+  auto name_section = getCustomSection("name");
+  if (name_section.size()) {
+    const byte_t *pos = name_section.data();
+    const byte_t *end = name_section.data() + name_section.size();
+    while (pos < end) {
+      if (*pos++ != 1) {
+        pos += parseVarint(pos, end);
+      } else {
+        const auto size = parseVarint(pos, end);
+        if (size == static_cast<uint32_t>(-1) || pos + size > end) {
+          function_names_index_ = {};
+          return;
+        }
+        const auto start = pos;
+        const auto namemap_vector_size = parseVarint(pos, end);
```

Comment:
By "same here" I meant that you need to verify the return value here (and in the two calls to `parseVarint` below).

Reply:
ah, sorry. I'm on it